device/pcie/pci_device.hpp

```diff
@@ -55,7 +55,8 @@ class PCIDevice {
     void write_tlb_reg(uint32_t byte_addr, std::uint64_t value_lower, std::uint64_t value_upper, std::uint32_t tlb_cfg_reg_size);
 
     void open_hugepage_per_host_mem_ch(uint32_t num_host_mem_channels);
-    tt::umd::architecture_implementation* get_architecture_implementation() const { return architecture_implementation.get(); }
+    bool reset_board();
+    tt::umd::TTDevice* get_architecture_implementation() const { return architecture_implementation.get(); }
```

Comment:
Does `get_architecture_implementation` still make sense?

Reply:
This will disappear with the next actual refactor, this was just a rename of the type so the following refactor is clearer. PCIDevice shouldn't hold TTDevice, but vice-versa, according to the redesign diagram.